Sources/ArcGISToolkit/Components/Offline/PreplannedListItemView.swift

```diff
@@ -55,6 +74,19 @@ struct PreplannedListItemView: View {
             .font(.body)
     }
     
+    @ViewBuilder private var deleteButton: some View {
+        if model.status.allowsRemoval {
+            Button("Delete") {
+                model.removeDownloadedPreplannedMapArea()
+                onDeletion()
```

Comment:
Since this PR already involves many changes, I might need to open a separate PR that targets this branch to address the feedback related to selection. I'll try to implement the following:

- If a map area is currently being viewed, it cannot be "selected" opened again
- A button to unselect the currently selected area that passes a `nil` value to the parent view
- A way to unselect a selected map that is being deleted

Reply:
Okay, I'll wait and see what you come up with.